client_code/_Components/MenuMixin.py

```diff
@@ -0,0 +1,115 @@
+from .._utils import fui
+from .MenuItem import MenuItem
+
+
+class MenuMixin():
+
```

Comment:
```suggestion
class MenuMixin():
    def __init__(self, component_node, menu_node):
        self._shown = False
        self._component_node = ...
        self._menu_node = ...
        self.add_event_handler("x-page-added", self._menu_mixin_mount)
        self.add_event_handler("x-page-removed", self._menu_mixin_cleanup)
```

then move the specific logic for fui into these
and call MenuMixin.__init__(...) inside the init methods of the other component